src/python/pants/backend/go/lint/golangci_lint/rules.py

```diff
@@ -110,15 +111,22 @@ async def run_golangci_lint(
         Get(GoModInfo, GoModInfoRequest(address)) for address in owning_go_mod_addresses
     )
 
+    # If cgo is enabled, golangci-lint needs to be able to locate the
+    # associated tools in its environment. This is injected in $PATH in the
+    # wrapper script.
+    cgo_tool_search_paths = (
+        ":".join(golang_env_aware.cgo_tool_search_paths) if golang_subsystem.cgo_enabled else ""
```

Comment:
This should request `GoBuildOptions` for the target and check `cgo_enabled` on the options instead so it takes into account the cgo status from `go_mod` and certain other targets (which can override the `golang_subsystem.cgo_enabled ` default setting).

Reply:
Would something like this work?
```python
    go_build_opts = await MultiGet(
        Get(GoBuildOptions, GoBuildOptionsFromTargetRequest(address)) for address in owning_go_mod_addresses
    )

    cgo_enabled = any(build_opts.cgo_enabled for build_opts in go_build_opts)
```

However I'm not sure about the behavior when multiple targets are selected. I think that usually you'd enable cgo for all targets or none of them, but this is just guesswork - I based my original solution on the existing code which just used `golang_subsystem.cgo_enabled` in the wrapper script to make sure the behavior stays the same.

Reply:
That code seems fine. `cgo_enabled` is only applicable in any event to `go_mod` and `go_binary` targets. Requesting options for all of the `go_package` targets to be linted will use whatever is defined on the relevant `go_mod` or in the default in configuration.

Reply:
Thanks for the info, fixed in 4d6f0d5261